packages/discord-utilities/src/lib/regexes.ts

```diff
@@ -0,0 +1,114 @@
+/**
+ * Regex that can capture the ID in Discord Channel mentions
+ * @raw `/^(?:<#)?(\d{17,19})>?$/`
+ * @remark Capture group 1 is the ID of the channel. It is named `id`.
+ */
+export const ChannelMentionRegex = /^(?:<#)?(?<id>\d{17,19})>?$/;
+
+/**
+ * Regex that matches links on the known Discord host names
+ * @raw `/^(?:\w+\.)?(?:discordapp.com|discord.gg|discord.com)$/i`
+ * @remark The regex is case insensitive
+ */
+export const DiscordHostnameRegex = /^(?:\w+\.)?(?:discordapp\.com|discord\.gg|discord\.com)$/i;
```

Comment:
This should:
1. Include ptb, canary
2. be optimized to not repeat `.com`, if applicable
3. Include named capture groups?
4. Why are you using `$` 🤔

Reply:
> Include named capture groups?

They are non-capture groups.

Reply:
Updated